src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
```

Comment:
PullProofAI

+     if (!numbers || !Array.isArray(numbers)) {
+       throw new Error('Input must be an array of numbers');
+     }
      if (!numbers.length) return 0;
```

#### Issue: Potential SQL Injection Vulnerability in getUserData Method
```diff


<!-- This is an auto-generated comment by PullProofAI -->